File: API/Features/Sales/BankAccounts/Dtos/BankAccountReadDto.cs
using API.Infrastructure.Classes;
using API.Infrastructure.Interfaces;

namespace API.Features.Sales.BankAccounts {

    public class BankAccountReadDto : IMetadata {

        // PK
        public int Id { get; set; }
        // FKs
        public int ShipOwnerId { get; set; }
        public int BankId { get; set; }
        // Fields
        public string Iban { get; set; }
        public bool IsActive { get; set; }
        // Metadata
        public string PostAt { get; set; }
        public string PostUser { get; set; }
        public string PutAt { get; set; }
        public string PutUser { get; set; }
        // Navigation
        public SimpleEntity ShipOwner { get; set; }
        public SimpleEntity Bank { get; set; }

    }

}

File: FrontEnd/src/app/features/sales/bankAccounts/classes/dtos/bankAccount-read-dto.ts
import { Metadata } from 'src/app/shared/classes/metadata'
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'

export interface BankAccountReadDto extends Metadata {

    // PK
    id: number
    // Object fields
    shipOwner: SimpleEntity
    bank: SimpleEntity
    // Fields
    iban: string
    isActive: boolean

}
